### PR TITLE
px4flow: sending OPTICAL_FLOW_RAD messages

### PR DESCRIPTION
**Pull request by okalachev**
PX4 [supports](https://github.com/PX4/Firmware/blob/81a80e0d5634c1796838d155abd73ab3ba0c2fd1/src/modules/mavlink/mavlink_receiver.cpp#L615) handling `OPTICAL_FLOW_RAD` messages.

Added an ability to send it using `OpitcalFlowRad.msg` (`px4flow/raw/send`). Can be useful for making an external optical flow system on a companion computer.

Coordinates are consistent with the receiving topic.

### CODE REVIEW

File: mavros_extras/src/plugins/px4flow.cpp [hunk 1]
```diff
@@ -149,6 +152,39 @@ class PX4FlowPlugin : public plugin::PluginBase {
...
+		flow_rad_msg.sensor_id = 0;
+		flow_rad_msg.integration_time_us = msg->integration_time_us;
+		flow_rad_msg.integrated_x = int_xy.x();
+		flow_rad_msg.integrated_y = int_xy.y();
+		flow_rad_msg.integrated_xgyro = int_gyro.x();
+		flow_rad_msg.integrated_ygyro = int_gyro.y();
+		flow_rad_msg.integrated_zgyro = int_gyro.z();
+		flow_rad_msg.temperature = msg->temperature;
+		flow_rad_msg.quality = msg->quality;
+		flow_rad_msg.temperature = msg->temperature * 100.0f; // temperature in centi-degrees Celsius
```
Comment (mhkabir) [inline mavros_extras/src/plugins/px4flow.cpp]: Temperature and quality are set twice.

Comment (okalachev) [reply mavros_extras/src/plugins/px4flow.cpp]: Thanks!

Comment (okalachev) [reply mavros_extras/src/plugins/px4flow.cpp]: FIxed.

Comment (vooon) [reply mavros_extras/src/plugins/px4flow.cpp]: That is why i prefer cog that things...

**Comment by okalachev** [comment]
@mhkabir, is it ok now?

File: mavros_extras/src/plugins/px4flow.cpp [hunk 2]
```diff
@@ -149,6 +152,37 @@ class PX4FlowPlugin : public plugin::PluginBase {
...
+	void send_cb(const mavros_msgs::OpticalFlowRad::ConstPtr msg)
+	{
+		mavlink::common::msg::OPTICAL_FLOW_RAD flow_rad_msg;
+
+		auto int_xy = ftf::transform_frame_baselink_aircraft(
+			Eigen::Vector3d(
+					msg->integrated_x,
+					msg->integrated_y,
+					0.0));
+		auto int_gyro = ftf::transform_frame_aircraft_baselink(
```
Comment (mhkabir) [inline mavros_extras/src/plugins/px4flow.cpp]: Why are the transforms for the gyro and the pixel flow different?

Comment (okalachev) [reply mavros_extras/src/plugins/px4flow.cpp]: Copy-paste issue again :(

**Comment by okalachev** [comment]
@mhkabir ?

File: mavros_extras/src/plugins/px4flow.cpp [hunk 3]
```diff
@@ -149,6 +152,37 @@ class PX4FlowPlugin : public plugin::PluginBase {
...
+					msg->integrated_x,
+					msg->integrated_y,
+					0.0));
+		auto int_gyro = ftf::transform_frame_baselink_aircraft(
+				Eigen::Vector3d(
+					msg->integrated_xgyro,
+					msg->integrated_ygyro,
+					msg->integrated_zgyro));
+
+		flow_rad_msg.time_usec = msg->header.stamp.toNSec() / 1000;
```
Comment (TSC21) [inline mavros_extras/src/plugins/px4flow.cpp]: Can you use cog here please?

**Pull request merged**